src/System.Net.NetworkInformation/tests/FunctionalTests/IPInterfacePropertiesTest.cs

```diff
@@ -0,0 +1,230 @@
+
+                Assert.NotNull(ipProperties.MulticastAddresses);
+                _log.WriteLine("- Multicast Addresses: " + ipProperties.MulticastAddresses.Count);
+                foreach (IPAddressInformation multi in ipProperties.MulticastAddresses)
+                {
+                    _log.WriteLine("-- " + multi.Address.ToString());
+                    _log.WriteLine("--- Dns Eligible: " + multi.IsDnsEligible);
+                    _log.WriteLine("--- Transient: " + multi.IsTransient);
+                }
+
+                Assert.NotNull(ipProperties.UnicastAddresses);
+                _log.WriteLine("- Unicast Addresses: " + ipProperties.UnicastAddresses.Count);
+                foreach (UnicastIPAddressInformation uni in ipProperties.UnicastAddresses)
+                {
+                    _log.WriteLine("-- " + uni.Address.ToString());
+                    _log.WriteLine("--- Preferred Lifetime: " + uni.AddressPreferredLifetime);
+                    _log.WriteLine("--- Valid Lifetime: " + uni.AddressValidLifetime);
+                    _log.WriteLine("--- Dhcp lease Lifetime: " + uni.DhcpLeaseLifetime);
+                    _log.WriteLine("--- Duplicate Address Detection State: " + uni.DuplicateAddressDetectionState);
+
+                    Assert.NotNull(uni.IPv4Mask);
+                    _log.WriteLine("--- IPv4 Mask: " + uni.IPv4Mask);
+                    _log.WriteLine("--- Dns Eligible: " + uni.IsDnsEligible);
+                    _log.WriteLine("--- Transient: " + uni.IsTransient);
+                    _log.WriteLine("--- Prefix Origin: " + uni.PrefixOrigin);
+                    _log.WriteLine("--- Suffix Origin: " + uni.SuffixOrigin);
+
+                    // Prefix Length
+                    _log.WriteLine("--- Prefix Length: " + uni.PrefixLength);
+                    Assert.NotEqual(0, uni.PrefixLength);
+                }
+
+                Assert.NotNull(ipProperties.WinsServersAddresses);
+                _log.WriteLine("- Wins Addresses: " + ipProperties.WinsServersAddresses.Count);
+                foreach (IPAddress wins in ipProperties.WinsServersAddresses)
+                {
+                    _log.WriteLine("-- " + wins.ToString());
+                }
+            }
+        }
+
+        [Fact]
+        public void IPInfoTest_AccessAllIPv4Properties_NoErrors()
+        {
+            foreach (NetworkInterface nic in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                _log.WriteLine("Nic: " + nic.Name);
+
+                IPInterfaceProperties ipProperties = nic.GetIPProperties();
+
+                _log.WriteLine("IPv4 Properties:");
+
+                if (!nic.Supports(NetworkInterfaceComponent.IPv4))
+                {
+                    NetworkInformationException nie = Assert.Throws<NetworkInformationException>(() =>
+                    {
+                        IPv4InterfaceProperties throwAway = ipProperties.GetIPv4Properties();
+                    });
```

Comment:
You don't actually need to store it into a local, and can just do:
```
var nie = Assert.Throws<NetworkInformationException>(() => ipProperties.GetIPv4Properties());
```
xunit has Throws overloads that take funcs rather than actions.